app/services/katello/candlepin_event_listener.rb

```diff
@@ -1,47 +1,24 @@
 module Katello
   class CandlepinEventListener
-    PROCESSED_COUNT_CACHE_KEY = 'candlepin_events_processed'.freeze
-    FAILED_COUNT_CACHE_KEY = 'candlepin_events_failed'.freeze
-    AMQP_QUEUE_NAME = 'event.org.candlepin.audit.AMQPBusPublisher'.freeze
+    STATUS_CACHE_KEY = 'candlepin_events_status'.freeze
 
     CandlepinEvent = Struct.new(:message_id, :subject, :content)
 
     @logger = ::Foreman::Logging.logger('katello/candlepin_events')
     @failed_count = 0
     @processed_count = 0
 
-    def self.start_service
-      loop do
-        begin
-          result = Katello::CandlepinListeningService.instance.start
-
-          break if result == :connected
-
-          @logger.info("Attempting to restart Candlepin Listening Service")
-          sleep 5
-        end
-      end
-    end
-
     def self.run
-      @thread.kill if @thread
+      initialize_listening_service
 
-      # run in own thread so connecting to qpid won't block the main process
-      @thread = Thread.new do
```

Comment:
Since the Monitor is running in a background thread we don't need to worry about blocking the main thread here any more, so I simplified this code